e2e/src/api/specs/search.e2e-spec.ts

```diff
@@ -0,0 +1,224 @@
+import { AssetFileUploadResponseDto, LoginResponseDto } from '@immich/sdk';
+import { readFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { Socket } from 'socket.io-client';
+import { errorDto } from 'src/responses';
+import { app, testAssetDir, utils } from 'src/utils';
+import request from 'supertest';
+import { afterAll, beforeAll, describe, expect, it } from 'vitest';
+
+const albums = { total: 0, count: 0, items: [], facets: [] };
+
+describe('/search', () => {
+  let admin: LoginResponseDto;
+  let assetFalcon: AssetFileUploadResponseDto;
+  let assetDenali: AssetFileUploadResponseDto;
+  let websocket: Socket;
+
+  beforeAll(async () => {
+    await utils.resetDatabase();
+    admin = await utils.adminSetup();
+    websocket = await utils.connectWebsocket(admin.accessToken);
+
+    const files: string[] = [
+      '/albums/nature/prairie_falcon.jpg',
+      '/formats/webp/denali.webp',
+      '/formats/raw/Nikon/D700/philadelphia.nef',
+      '/albums/nature/orychophragmus_violaceus.jpg',
+      '/albums/nature/notocactus_minimus.jpg',
+      '/albums/nature/silver_fir.jpg',
+      '/albums/nature/tanners_ridge.jpg',
+      '/albums/nature/cyclamen_persicum.jpg',
+      '/albums/nature/polemonium_reptans.jpg',
+      '/albums/nature/wood_anemones.jpg',
+      '/formats/heic/IMG_2682.heic',
+      '/formats/jpg/el_torcal_rocks.jpg',
+      '/formats/png/density_plot.png',
+      '/formats/motionphoto/Samsung One UI 6.jpg',
+      '/formats/motionphoto/Samsung One UI 6.heic',
+      '/formats/motionphoto/Samsung One UI 5.jpg',
+      '/formats/raw/Nikon/D80/glarus.nef',
+      '/metadata/gps-position/thompson-springs.jpg',
+    ];
+    const assets: AssetFileUploadResponseDto[] = [];
+    for (const filename of files) {
+      const bytes = await readFile(join(testAssetDir, filename));
+      assets.push(
+        await utils.createAsset(admin.accessToken, {
+          deviceAssetId: `test-${filename}`,
+          assetData: { bytes, filename },
+        }),
+      );
+    }
+
+    for (const asset of assets) {
+      await utils.waitForWebsocketEvent({ event: 'upload', assetId: asset.id });
+    }
```

Comment:
Don't you want something like
```ts
await Promise.all(assets.map((asset) => utils.waitForWebsocketEvent({ event: 'upload', assetId: asset.id })));
```
instead? With the current approach this may wait for an event that already happened, no?

Reply:
It works even if the event has already happened. I find mine more readable and there is no performance benefit with running them in parallel since they're just waiting either way.